compiler-cli/src/build_lock.rs

```diff
@@ -9,11 +10,21 @@ pub(crate) struct BuildLock {
 
 // TODO: return errors rather than crashing.
 impl BuildLock {
-    pub fn new() -> Result<Self> {
+    pub fn new_scoped(mode: &Mode, target: &Target) -> Result<Self> {
```

Comment:
These implement `Copy` so can pass them in rather than as references

Reply:
done